internal/locate/region_request.go

```diff
@@ -1183,6 +1183,18 @@ func (s *RegionRequestSender) sendReqToRegion(bo *retry.Backoffer, rpcCtx *RPCCo
 		sendToAddr = rpcCtx.ProxyAddr
 	}
 
+	// Count the replica number as the RU cost factor.
+	req.ReplicaNumber = 1
```

Comment:
I think we need to optimize here. The write node number should be calculated for once instead of calculating every time (Though some node role might change). The overhead is too much...

Reply:
The loop is simple enough, so it seems not big overhead to me. A benchmark should tell.

Reply:
Before:

```shell
BenchmarkSendReqToRegion/replicaNum_3-8         	  440764	      2661 ns/op	    2075 B/op	      49 allocs/op
BenchmarkSendReqToRegion/replicaNum_5-8         	  416978	      2887 ns/op	    2396 B/op	      54 allocs/op
BenchmarkSendReqToRegion/replicaNum_50-8        	  180320	      6511 ns/op	    8211 B/op	     147 allocs/op
BenchmarkSendReqToRegion/replicaNum_100-8       	  108415	     10799 ns/op	   14886 B/op	     248 allocs/op
BenchmarkSendReqToRegion/replicaNum_1000-8      	   15418	     78474 ns/op	  139241 B/op	    2053 allocs/op
```

After:

```shell
BenchmarkSendReqToRegion/replicaNum_3-8         	  438793	      2678 ns/op	    2072 B/op	      49 allocs/op
BenchmarkSendReqToRegion/replicaNum_5-8         	  416216	      2836 ns/op	    2396 B/op	      54 allocs/op
BenchmarkSendReqToRegion/replicaNum_50-8        	  175693	      6606 ns/op	    8219 B/op	     147 allocs/op
BenchmarkSendReqToRegion/replicaNum_100-8       	   96657	     10785 ns/op	   14921 B/op	     248 allocs/op
BenchmarkSendReqToRegion/replicaNum_1000-8      	   15133	     79340 ns/op	  139246 B/op	    2053 allocs/op
```

Reply:
I am fine with it now.